server/routes/soonBusStop.js

```diff
@@ -0,0 +1,126 @@
+import { calcCWorCCW } from "./direction.js"
+const busStops = require('./bus-stops.json')
+const defaultDatabase = require('./firebase.js')
+
+// updates next 3 bus stops for every bus
+export function nextBusStops() {
+  const busCollection = [];
+  let stops_arr_CW = [];
+  let stops_arr_CCW = [];
+
+  // Push all bus ID's into array
+  defaultDatabase.collection('busses').get().then(function(querySnapshot) {
+    querySnapshot.forEach(doc => {
+      // Store the busses into busCollection as an array of objects
+      busCollection.push ({
+        latitude: doc.data().lastLatitude,
+        longitude: doc.data().lastLongitude,
+        previousLocationArray: doc.data().previousLocationArray
+      });
+    });
+  });
+
+  // Initialize CW array with false values
+  const cwObj = busStops.bstop.CW;
+  for (let i = 0; i < cwObj.length; i++) {
+    let locationName = Object.keys(cwObj[i])[0];
+    stops_arr_CW.push({
+      id: locationName,
+      value: false
+    })
+  }
+
+  // Initialize CCW array with false values
+  const ccwObj = busStops.bstop.CCW;
```

Comment:
Minor optimization could be made by simply moving CW and CCW to the first level of the busStops json:
eg.
{
 CW:...,
 CCW:...
}

But if it breaks too much stuff dont have to change